solr/core/src/test/org/apache/solr/search/QueryEqualityTest.java

```diff
@@ -1190,14 +1190,8 @@ public void testCompares() throws Exception {
     assertFuncEquals("gte(foo_i,2)", "gte(foo_i,2)");
     assertFuncEquals("eq(foo_i,2)", "eq(foo_i,2)");
 
-    boolean equals = false;
-    try {
-      assertFuncEquals("eq(foo_i,2)", "lt(foo_i,2)");
-      equals = true;
-    } catch (AssertionError e) {
-      //expected
-    }
-    assertFalse(equals);
+    expectThrows(AssertionError.class, "expected error, functions are not equal",
```

Comment:
[0] Not suggesting you change it here, but....kindof weird that there's just not an `assertFuncNotEquals`